actions/appConfig.js

```diff
@@ -1,35 +1,76 @@
-* ___________________
-*
-* Copyright 2023 Adobe
-* All Rights Reserved.
-*
-* NOTICE: All information contained herein is, and remains
-* the property of Adobe and its suppliers, if any. The intellectual
-* and technical concepts contained herein are proprietary to Adobe
-* and its suppliers and are protected by all applicable intellectual
-* property laws, including trade secret and copyright laws.
-* Dissemination of this information or reproduction of this material
-* is strictly forbidden unless prior written permission is obtained
-* from Adobe.
-************************************************************************* */
+/* ***********************************************************************
+ * ADOBE CONFIDENTIAL
+ * ___________________
+ *
+ * Copyright 2023 Adobe
+ * All Rights Reserved.
+ *
+ * NOTICE: All information contained herein is, and remains
+ * the property of Adobe and its suppliers, if any. The intellectual
+ * and technical concepts contained herein are proprietary to Adobe
+ * and its suppliers and are protected by all applicable intellectual
+ * property laws, including trade secret and copyright laws.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe.
+ ************************************************************************* */
+
+const crypto = require('crypto');
 
 class AppConfig {
-    appConfig = {};
+    configMap = {};
 
     setAppConfig(params) {
-        this.appConfig.fgSite = params.fgSite;
-        this.appConfig.fgClientId = params.fgClientId;
-        this.appConfig.fgAuthority = params.fgAuthority;
-        this.appConfig.shareUrl = params.shareUrl;
-        this.appConfig.fgShareUrl = params.fgShareUrl;
-        this.appConfig.rootFolder = params.rootFolder;
-        this.appConfig.fgRootFolder = params.fgRootFolder;
+        this.configMap.fgSite = params.fgSite;
+        this.configMap.fgClientId = params.fgClientId;
+        this.configMap.fgAuthority = params.fgAuthority;
+        this.configMap.shareUrl = params.shareUrl;
+        this.configMap.fgShareUrl = params.fgShareUrl;
+        this.configMap.rootFolder = params.rootFolder;
+        this.configMap.fgRootFolder = params.fgRootFolder;
+        this.configMap.clientId = params.clientId;
+        this.configMap.tenantId = params.tenantId;
+        this.configMap.driveId = params.driveId;
+        this.configMap.certPassword = params.certPassword;
+        this.configMap.certKey = params.certKey;
+        this.configMap.certThumbprint = params.certThumbprint;
+        this.configMap.testAccessToken = params.testAccessToken;
+        this.configMap.skipInProg = (params.skipInProgressCheck || '').toLowerCase() === 'true';
+        this.extractPrivateKey();
+        // Test token in base64
+        if (this.configMap.testAccessToken) {
+            this.configMap.testAccessToken = Buffer.from(this.configMap.testAccessToken, 'base64');
```

Comment:
As discussed, let's not have this option. Let all FG actions use the app based token.